packages/shared/src/hooks/usePrompt.ts

```diff
@@ -0,0 +1,53 @@
+import { useQuery, useQueryClient } from 'react-query';
+import { ReactNode } from 'react';
+import { ButtonProps, StyledButtonProps } from '../components/buttons/Button';
+
+export const PROMPT_KEY = 'prompt';
+
+export type PromptButtonProps = StyledButtonProps &
+  Omit<ButtonProps<'button'>, 'onClick'> & {
+    title?: string;
+  };
+
+export type PromptOptions = {
+  title: string;
+  description?: string | ReactNode;
+  okButton?: PromptButtonProps;
+  cancelButton?: PromptButtonProps;
+};
+
+type Prompt = {
+  options: PromptOptions | null;
+  onSuccess: () => void;
+  onFail: () => void;
+};
+
+type UsePromptRet = {
+  showPrompt: (options: PromptOptions) => Promise<boolean>;
+  prompt: Prompt;
+};
+
+export function usePrompt(): UsePromptRet {
+  const client = useQueryClient();
+  const { data: prompt } = useQuery<Prompt>(PROMPT_KEY);
+  const setPrompt = (data: Prompt) => client.setQueryData(PROMPT_KEY, data);
```

Comment:
Can you test what happens if there is a case where there are 2 different prompts after each other?
So prompt 1 will prompt another one?

Reply:
It would replace prompt 1, I can add a stack if there is such a use-case. I never saw 2 consequent prompts at the same time.

Reply:
Alright, not at the same time, but in consecutive order, was wondering if react query handled it correct with the same key, but no worries don't think we have it at the moment.